client/me/purchases/billing-history/test/filter-transactions.ts

```diff
@@ -0,0 +1,471 @@
+				variation: 'Variation1',
+			},
+		],
+	},
+	{
+		...mockTransaction,
+		date: '2018-04-11T13:11:27',
+		service: 'WordPress.com',
+		cc_name: 'name2',
+		cc_type: 'visa',
+		items: [
+			{
+				...mockItem,
+				amount: '$5.75',
+				type: 'new purchase',
+				variation: 'Variation2',
+			},
+			{
+				...mockItem,
+				amount: '$8.00',
+				type: 'new purchase',
+				variation: 'Variation2',
+			},
+		],
+	},
+	{
+		...mockTransaction,
+		date: '2018-03-11T21:00:00',
+		service: 'Store Services',
+		cc_name: 'name1 surname1',
+		cc_type: 'visa',
+		items: [
+			{
+				...mockItem,
+				amount: '$3.50',
+				type: 'new purchase',
+				variation: 'Variation2',
+			},
+			{
+				...mockItem,
+				amount: '$5.00',
+				type: 'new purchase',
+				variation: 'Variation2',
+			},
+		],
+	},
+	{
+		...mockTransaction,
+		date: '2018-03-15T10:39:27',
+		service: 'Store Services',
+		cc_name: 'name2',
+		cc_type: 'mastercard',
+		items: [
+			{
+				...mockItem,
+				amount: '$4.86',
+				type: 'new purchase',
+				variation: 'Variation1',
+			},
+			{
+				...mockItem,
+				amount: '$1.23',
+				type: 'new purchase',
+				variation: 'Variation1',
+			},
+		],
+	},
+	{
+		...mockTransaction,
+		date: '2018-03-13T16:10:45',
+		service: 'WordPress.com',
+		cc_name: 'name1 surname1',
+		cc_type: 'visa',
+		items: [
+			{
+				...mockItem,
+				amount: '$3.50',
+				type: 'new purchase',
+				variation: 'Variation1',
+			},
+		],
+	},
+	{
+		...mockTransaction,
+		date: '2018-01-10T14:24:38',
+		service: 'WordPress.com',
+		cc_name: 'name2',
+		cc_type: 'mastercard',
+		items: [
+			{
+				...mockItem,
+				amount: '$4.20',
+				type: 'new purchase',
+				variation: 'Variation2',
+			},
+		],
+	},
+	{
+		...mockTransaction,
+		date: '2017-12-10T10:30:38',
+		service: 'Store Services',
+		cc_name: 'name1 surname1',
+		cc_type: 'visa',
+		items: [
+			{
+				...mockItem,
+				amount: '$3.75',
+				type: 'new purchase',
+				variation: 'Variation1',
+			},
+		],
+	},
+	{
+		...mockTransaction,
+		date: '2017-12-01T07:20:00',
+		service: 'Store Services',
+		cc_name: 'name1 surname1',
+		cc_type: 'visa',
+		items: [
+			{
+				...mockItem,
+				amount: '$9.50',
+				type: 'new purchase',
+				variation: 'Variation1',
+			},
+		],
+	},
+	{
+		...mockTransaction,
+		date: '2017-11-24T05:13:00',
+		service: 'WordPress.com',
+		cc_name: 'name1 surname1',
+		cc_type: 'visa',
+		items: [
+			{
+				...mockItem,
+				amount: '$8.40',
+				type: 'new purchase',
+				variation: 'Variation2',
+			},
+		],
+	},
+	{
+		...mockTransaction,
+		date: '2017-01-01T00:00:00',
+		service: 'Store Services',
+		cc_name: 'name2',
+		cc_type: 'mastercard',
+		items: [
+			{
+				...mockItem,
+				amount: '$2.40',
+				type: 'new purchase',
+				variation: 'Variation1',
+			},
+		],
+	},
+];
+
+const state = {
+	billingTransactions: {
+		items: {
+			past,
+		},
+		ui: { past: undefined, upcoming: undefined },
+	},
+};
+
+describe( 'filterTransactions()', () => {
```

Comment:
Despite looking quite different now, these tests are functionally the same as they were when they were in `get-filtered-billing-transactions`.